functions/v2/test/index.test.js

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const assert = require('assert');
+const {request} = require('gaxios');
+const {exec} = require('child_process');
+const waitPort = require('wait-port');
+
+const startFF = async (target, signature, port) => {
```

Comment:
suggestion: I tend to feel system tests using the function framework locally have a high overhead relative to their advantage over unit tests. Would kicking off v2 samples be a good moment to switch to Deploy tests?

Reply:
I believe the Functions Frameworks are directly integrated into [all versions of] GCF, so this will give us a medium-high fidelity test.

I agree with your sentiment, but I think we need to be more organized/deliberate about how we add deployment tests. (In particular, do we want to write deployment tests in every language - or have some common "translation" layer so we only need write them in one language [e.g. PHP, which already has deployment tests]?)